src/units/unit.cpp

```diff
@@ -1789,11 +1789,8 @@ bool unit::resistance_filter_matches(const config& cfg, bool attacker, const std
 	return true;
 }
 
-int unit::resistance_against(const std::string& damage_name,bool attacker,const map_location& loc, const_attack_ptr weapon, const_attack_ptr opp_weapon) const
+int unit::resistance_abilitie(unit_ability_list resistance_abilities, const std::string& damage_name, bool attacker, int res) const
 {
-	int res = opp_weapon ? movement_type_.resistance_against(*opp_weapon) : movement_type_.resistance_against(damage_name);
```

Comment:
This removes a call to `movetype::resistance_against(attack_type const&) const`, which is good.

I'm wondering if it should go further, and remove all knowledge of `attack_type` from `movetype.hpp`, so that movement types only support `resistance_against(const std::string & damage_type) const`. Otherwise I think this logic about working out which weapon would be used would have to be duplicated there, and I don't want duplicated code.

Reply:
in ai/formula/function_table.cpp line 72 movement_type().resistance_against(attack) is called, and perhaps what have damage_type() remain pertinent in this case.

Reply:
Will do a followup once this is merged.